File: Support/old c++ code/global.cpp

#include "lota.h"


/****************************************************************************
 *  Global::Global()														*
 *																			*
 *  This constructor initializes all the values in the global				*
 *	class to their default values.											*
 ****************************************************************************/
Global::Global()
{
	int i;
	
	locked = false;
	theCursor = 'A';
	done = false;

	for (i = 0; i < 5; i++)
	{

		bottomColor[i] = new unsigned int[39];
		memset(bottomColor[i], CreateRGB(255,255,255), sizeof(bottomColor));

	}

	myApp = NULL;
	hWnd = NULL;

	pTitleScreen = NULL;

	player.FaceDirection(lotaEast);

	timerSet = false;
	stdDisplay = false;
	quickMenu = false;

	newGraphics = false;

	raftFacing = lotaEast;
	charAnimCount = 0;
	d3dViewport = false;
	ZBufferEnable = true;
	pJoystick = NULL;
	LeftMenuActive = false;
	HPColor = 0x00FFFFFF;
	allowEnter = true;

	disableEncounters = false;

	walkTime = 75;

	invisible = false;
	guard = false;

	titleScreen = true;
	titleState = stNoState;
	titleMenu = 0;

}

/****************************************************************************
 *  Global::~Global()														*
 *																			*
 *  This destructor deletes all dynamically allocated memory for the global	*
 *	class.																	*
 ****************************************************************************/
Global::~Global()
{

	for (int i = 0; i < 5; i++)
	{
		delete [] bottomColor[i];
	}

	Unlock();
	ReleaseFont();

}

/****************************************************************************
 *  void Global::SetHInstance(HINSTANCE instance)							*
 *																			*
 *  Here the hInstance value is stored for hInstance to return it when		*
 *	called.																	*
 ****************************************************************************/
void Global::SetHInstance(HINSTANCE instance, HWND window)
{
	if (locked == false)
	{
		char	tempSpace[40];
		int		i;

		myApp = instance;
		hWnd = window;

		for (i = 0; i < 8; i++)
		{
			LoadString (hInstance(), i + 1, tempSpace, 39);
			weaponName[i] = tempSpace;
		}

		for (i = 0; i < 5; i++)
		{
			LoadString (hInstance(), i + 9, tempSpace, 39);
			armorName[i] = tempSpace;

		}

		for (i = 0; i < 5; i++)
		{
			LoadString (hInstance(), i + 14, tempSpace, 39);
			qualityName[i] = tempSpace;

		}
	}
}

/****************************************************************************
 *  LPDIRECTDRAWSURFACE7 Global::Font()										*
 *																			*
 *  This function returns the DirectDraw surface that the font bitmap is	*
 *	stored on.																*
 ****************************************************************************/
LPDIRECTDRAWSURFACE7 Global::Font()
{
	return myFont;
}


/****************************************************************************
 *  LPDIRECTDRAWSURFACE7 Global::Tiles()									*
 *																			*
 *  This function returns the DirectDraw surface that the tiles bitmap is	*
 *	stored on.																*
 ****************************************************************************/
LPDIRECTDRAWSURFACE7 Global::Tiles()
{
	return myTiles;
}

/****************************************************************************
 *  LPDIRECTDRAWSURFACE7 Global::Character()								*
 *																			*
 *  This function returns the DirectDraw surface that the tiles bitmap is	*
 *	stored on.																*
 ****************************************************************************/
LPDIRECTDRAWSURFACE7 Global::Character()
{
	return myCharacter;
}

/****************************************************************************
 *  LPDIRECTDRAWSURFACE7 Global::Monsters()									*
 *																			*
 *  This function returns the DirectDraw surface that the tiles bitmap is	*
 *	stored on.																*
 ****************************************************************************/
LPDIRECTDRAWSURFACE7 Global::Monsters()
{
	return pOverlandMonsters;
}

/****************************************************************************
 *  bool Global::LoadFont()													*
 *																			*
 *  This function loads the font from our file (our resource, when that's	*
 *	properly implemented) and returns a value as to whether or not the font	*
 *	was successfully loaded.												*
 ****************************************************************************/
bool Global::LoadFont()
{
	int	tiles;

	if (locked == false)
	{
		ReleaseFont();

		DDSetColorKey (g_pDDSTemp, RGB(0,0,0));
		myFont = DDLoadBitmap(g_pDD, MAKEINTRESOURCE(LOTA_FONT), 0, 0);

		if (!myFont)
		{
			return false;
		}

		DDSetColorKey (myFont, RGB(255,255,255));
	
		switch (map.TileSet())
		{
		case 0:
			tiles = LOTA_TILES;
			break;
		case 1:
			tiles = LOTA_TOWNTILES;
			break;
		case 2:
			tiles = LOTA_CASTLETILES;
			break;
		default:
			tiles = LOTA_TILES;
			break;

		}

		myTiles = DDLoadBitmap(g_pDD, MAKEINTRESOURCE(tiles), 0, 0);

		if (!myTiles)
		{
			return false;
		}

		myCharacter = DDLoadBitmap(g_pDD, MAKEINTRESOURCE(LOTA_CHARACTER), 0, 0);
		
		if (!myCharacter)
		{
			return false;
		}

		pOverlandMonsters = DDLoadBitmap(g_pDD, MAKEINTRESOURCE(LOTA_OVERLANDMONSTERS), 0, 0);
		
		if (!myCharacter)
		{
			return false;
		}

		pTitleScreen = DDLoadBitmap(g_pDD, MAKEINTRESOURCE(LOTA_TITLE), 0, 0);
		if (!pTitleScreen)
			return false;

		DDSetColorKey (myCharacter, RGB(0,0,0));
		DDSetColorKey (pOverlandMonsters, RGB(0,0,0));

		return true;
	}
	else
	{
		return false;
	}
}

/****************************************************************************
 *  HINSTANCE Global::hInstance()											*
 *																			*
 *  This function returns the handle to the program instance that was		*
 *	passed to the program when it was first run.							*
 ****************************************************************************/
HINSTANCE Global::hInstance()
{
	return myApp;
}

/****************************************************************************
 *	HWND Global::hwnd()														*
 *																			*
 *  This function returns the handle to the window.							*
 ****************************************************************************/
HWND Global::hwnd()
{
	return hWnd;
}


/****************************************************************************
 *  void Global::Lock()			and			void Global::Unlock()			*
 *																			*
 *  These two functions toggle locked, which is a value that allows the		*
 *	program to modify key values, like hInstance and myFont.  Once the		*
 *	values are set, the program calls lock and then those values may not	*
 *	modified until unlock has been called.									*
 ****************************************************************************/
void Global::Lock()
{
	locked = true;
}

void Global::Unlock()
{
	locked = false;
}

/****************************************************************************
 *  char Global::Menu(int mx, int my)										*
 *																			*
 *  This function will return a single character value at an (x, y) point	*
 *	in the menu at the left.												*
 ****************************************************************************/
char Global::Menu(int mx, int my)
{
	return menuArray[my][mx];
}

/****************************************************************************
 *  char Global::Menu(int my)												*
 *																			*
 *  This function returns a pointer to an entire line from the menu at the 	*
 *	left.																	*
 ****************************************************************************/
String Global::Menu(int my)
{
	String line = menuArray[my];

	return line;
}

/****************************************************************************
 *  String Global::Menu(char)												*
 *																			*
 *  This returns an entire line from the menu that matches the first		*
 *	character of the command selected.  If it's passed						*
 *  a value that's not in the menu, it returns NULL							*
 ****************************************************************************/
String Global::Menu(char newCursor)
{

	int i;
	String csr;

	newCursor = toupper(newCursor);

	for (i = 0; i < 14; i++)
	{
		if (menuArray[i][0] == newCursor)
		{
			csr = menuArray[i];

			return csr;

		}

	}

	return "";

}
/****************************************************************************
 *  void Global::WriteMenu(int loc, const String line)						*
 *																			*
 *  This function allows the program to set a line in the menu.				*
 ****************************************************************************/
void Global::WriteMenu(int loc, const String line)
{
	if (loc < 17)
		menuArray[loc] = line;
}

/****************************************************************************
 *  char Global::cursor()													*
 *																			*
 *  This will return the first character of the last command entered.		*
 ****************************************************************************/
char Global::cursor()
{
	return theCursor;
}

/****************************************************************************
 *  char Global::cursor(char newCursor)										*
 *																			*
 *  This will return the first character of the command selected.  It also	*
 *	allows setting the command with the argument.  If a matching command is *
 *  found, then it will set the cursor at that command and return that		*
 *  value.  If not, it will return the last selected value.	 If it's passed	*
 *  a value that's not in the menu, it returns -1							*
 ****************************************************************************/
char Global::cursor (char newCursor)
{
	int i;

	newCursor = toupper(newCursor);

	if (newCursor != 0)
	{

		for (i = 0; i < 14; i++)
		{
			if (menuArray[i][0] == newCursor)
			{
				theCursor = newCursor;
				return theCursor;

			}

		}
	}

	return -1;

}

/****************************************************************************
 *  int Global::cursorPos ()												*
 *																			*
 *  This will return the numerical position of the cursor so that it may 	*
 *	be plotted on the screen.												*	
 ****************************************************************************/
int Global::cursorPos (int change)
{
	int i;
	bool found = false;

	do
	{
		theCursor += change;

		if (theCursor < 'A')
			theCursor = 'Z';
		
		if (theCursor > 'Z')
			theCursor = 'A';

		for (i = 0; i < 14; i++)
		{
			if (menuArray[i][0] == theCursor)
			{
				found = true;
				return i;

			}
		}
	} while (!found && change != 0);

	return 0;
	
}

/****************************************************************************
 *  void Global::AddBottom(const char *line, const int *colors)				*
 *																			*
 *  This function adds a line to the bottom of the action window and		*
 *	scrolls the remaining lines up one each.								*	
 ****************************************************************************/
void Global::AddBottom(const String& line, const unsigned int color)
{
	unsigned int colors[40];

	for (int i = 0; i < 40; i++)
	{
		colors[i] = color;
	}

	AddBottom(line, colors);
}

void Global::AddBottom(const String& line, const unsigned int *colors)
{
	int i;

	bottom[4] = "";
	delete [] bottomColor[4];

	for (i = 3; i >= 0; i--)
	{
		bottom[i + 1] = bottom[i];
		bottomColor[i + 1] = bottomColor[i];
	}


	bottomColor[0] = new unsigned int[39];

	bottom[0] = line;

	if (colors)
	{
		for (i = 0; i < 39; i++)
		{
			bottomColor[0][i] = colors[i];
		}
	}
	else
	{
		unsigned int fontColor = lotaWhite;

		if (map.MapType() == mapDungeon)
		{
			fontColor = lotaCyan;
		}
		for (i = 0; i < 39; i++)
		{
			bottomColor[0][i] = fontColor;
		}
	}

}

void Global::WriteSlow(const String& line, int loc, const unsigned int color)
{
	int i = 0;
	unsigned int colors[40];
	String temp;

	while (i < len(line) && i < 40)
	{
		colors[i++] = color;
		temp = left(line, i);
		
		UpdateBottom(temp, loc, colors);

		wait(50);
	}

}

/****************************************************************************
 *  void Global::UpdateBottom(const char *line, int loc = 0)				*
 *																			*
 *  This function updates a line in the action window.						*
 ****************************************************************************/
void Global::UpdateBottom(const String& line, int loc, const unsigned int * colors)
{

	bottom[loc] = line;

	if (colors)
	{
		for (int i = 0; i < 39; i++)
		{
			bottomColor[loc][i] = colors[i];
		}

	}

	
}

void Global::UpdateBottom(const String& line, unsigned int color)
{
	unsigned int colors[40];

	for (int i = 0; i < 40; i++)
	{
		colors[i] = color;
	}

	UpdateBottom(line, 0, colors);

}

/****************************************************************************
 *  char* Global::Bottom(int line)											*
 *																			*
 *  This function returns a line from the bottom of the action window.		*
 *	Returns null if the line was out of range.								*	
 ****************************************************************************/
String Global::Bottom(int line)
{
	String tempspace;

	if (line >= 0 && line <=4 )
	{

		tempspace = bottom[line];

		return tempspace;
	}
	else
	{
		return NULL;
	}

}

/****************************************************************************
 *  int * Global::BottomColor(int line)										*
 *																			*
 *  This function returns a line from the bottom of the action window.		*
 *	Returns null if the line was out of range.								*	
 ****************************************************************************/
unsigned int * Global::BottomColor(int line)
{
	static unsigned int tempspace[39];
	int i;

	if (line >= 0 && line <=4 )
	{

		for (i = 0; i < 39; i++)
		{
			tempspace[i] = bottomColor[line][i];
		}

		return tempspace;
	}
	else
	{
		return NULL;
	}

}

void Global::ClearBottom()
{
	int i;

	for (i = 0; i < 5; i++)
	{
		AddBottom("");
	}

}

/****************************************************************************
 *  void Global::ResetTimers()												*
 *																			*
 *  Resets the timers for use with commands									*	
 ****************************************************************************/
void Global::ResetTimers()
{
	DestroyTimers();

	SetTimer (hWnd, commandTimer, waitCommand, NULL);
	SetTimer (hWnd, passTimer, 5000 + 750 * player.Gamespeed(), NULL);

	timerSet = true;
}

/****************************************************************************
 *  void Global::ResetTimers()												*
 *																			*
 *  Destroys the timers for use with commands so the window proc won't be	*
 *	interrupted.															*
 ****************************************************************************/
void Global::DestroyTimers()
{
	if (timerSet == true)
	{
		KillTimer (hWnd, passTimer);
		KillTimer (hWnd, commandTimer);
	}

	timerSet = false;
}

void Global::ReleaseFont()
{
	/*
	LPDIRECTDRAWSURFACE7		myFont;				// stores the handle to the font
	LPDIRECTDRAWSURFACE7		myTiles;			// stores the handle to the tiles
	LPDIRECTDRAWSURFACE7		myCharacter;		// stores the handle to the character sprites
	LPDIRECTDRAWSURFACE7		pOverlandMonsters;	// stores the handle to the overland monster sprites

	LPDIRECTDRAWSURFACE7	wallTexture;			// stores the pointer to the wall texture
	LPDIRECTDRAWSURFACE7	floorTexture;			// stores the pointer to the floor texture
	LPDIRECTDRAWSURFACE7	ceilingTexture;			// stores the pointer to the ceiling texture
	LPDIRECTDRAWSURFACE7	floorHoleTexture;		// stores the pointer to the floorhole texture

	LPDIRECTINPUTDEVICE2	pJoystick;				// Joystick

	*/
	if (locked == false)
	{
		if (myFont != NULL)
		{
			myFont->Release();
			myFont = NULL;
		}
		if (myTiles != NULL)
		{
			myTiles->Release();
			myTiles = NULL;
		}
		if (myCharacter != NULL)
		{
			myCharacter->Release();
			myCharacter = NULL;
		}
		if (pOverlandMonsters != NULL)
		{
			pOverlandMonsters->Release();
			pOverlandMonsters = NULL;
		}
		if (wallTexture != NULL)
		{
			wallTexture->Release();
			wallTexture = NULL;
		}
		if (floorTexture != NULL)
		{
			floorTexture->Release();
			floorTexture = NULL;
		}
		if (ceilingTexture != NULL)
		{
			ceilingTexture->Release();
			ceilingTexture = NULL;
		}
		if (floorHoleTexture != NULL)
		{
			floorHoleTexture->Release();
			floorHoleTexture = NULL;
		}
		if (pTitleScreen != NULL)
		{
			pTitleScreen->Release();
			pTitleScreen = NULL;
		}


	}
}

int Global::AnimFrame(int frame)
{
	static int last = 0;

	if (animating)
	{
		
		if (frame == -1)
		{
			return animFrame;
		}
		else if (last + 150 <= clock())
		{
			
			if (frame == animInc)
			{
				animFrame++;
				
				if (animFrame > 2)
				{
					animFrame = 0;
				}
			} 
			else if (frame == animDec)
			{
				animFrame--;
				
				if (animFrame < 0)
				{
					animFrame = 2;
				}
			}
			else if (frame >= 0 && frame <= 2)
			{
				animFrame = frame;
			}
			
			last = clock();
			
			return animFrame;
		}
	}

	return 0;

}

bool Global::Animating()
{
	if (animating == false)
	{
		animFrame = 0;
	}

	return animating;

}

bool Global::Animating(bool an)
{

	animating = an;

	if (an == false)
	{
		animFrame = 0;
	}
	else
	{
	}

	return animating;
}

// Returns the default font color
unsigned int Global::DefaultColor()
{
	if (map.MapType() == mapDungeon)
	{
		return lotaCyan;
	}
	
	return lotaWhite;

}
String Global::WeaponName(int a)
{
	return weaponName[a - 1];
}
int Global::WeaponCost(int w, int q)
{
	return 	int(9.639302862 + 12.709725901 * w + 7.448174718 * pow(w, 2) + 
			5.552075007 * q + 0.731199405 * pow(q, 2) + 4.290595417 * w * q);

}
String Global::ArmorName(int a)
{
	return armorName[a - 1];
}
int Global::ArmorCost(int a, int q)
{
	return 120 * a + 12 * q + 8 * q * a;
}
String Global::QualityName(int a)
{
	return qualityName[a];
}

File: Xle/Xle/Global.cs
using System;
using System.Collections.Generic;
using System.Text;

using AgateLib;
using AgateLib.DisplayLib;
using AgateLib.InputLib;
using AgateLib.Geometry;

namespace ERY.Xle
{

    // Directional definitions
    public enum Direction
    {
        None = 0,
        East = 1,
        North,
        West,
        South
    }

    // Animation definitions
    [Obsolete]
    public enum AnimType
    {
        Invalid = -1,
        animInc = -2,
        animDec = -3,
    }
    // Terrain Definitions
    public enum TerrainType
    {
        All = -1,
        Water,
        Mountain,
        Grass,
        Forest,
        Desert,
        Swamp,
        Mixed,
        Foothills,
        Scrubland,
    }


    public static class g
    {
        static int userControl;
        static int mode;
        static int animFrame;
        static bool animating;

        static FontSurface myFont;		// stores the handle to the font
        static Surface myTiles;			// stores the handle to the tiles
        static Surface myCharacter;		// stores the handle to the character sprites
        static Surface pOverlandMonsters;	// stores the handle to the overland monster sprites

        static string[] bottom = new string[5];			// keeps the bottom portion of the screen
        static Color[][] bottomColor = new Color[5][];	// keeps the bottom colors on the screen
        static string[] weaponName = new string[20];		// stores weapon names
        static string[] armorName = new string[20];		// stores armor names
        static string[] qualityName = new string[6];		// stores quality names


        // constructors and destructor:
        static g()
        {
            
            done = false;

            for (int i = 0; i < 5; i++)
            {
                bottomColor[i] = new Color[40];

                for (int j = 0; j < bottomColor[i].Length; j++)
                    bottomColor[i][j] = XleColor.White;

            }
            
            stdDisplay = 0;

            newGraphics = false;

            raftFacing = Direction.East;
            charAnimCount = 0;
            d3dViewport = false;
            ZBufferEnable = true;
            LeftMenuActive = false;
            HPColor = XleColor.White;
            allowEnter = true;

            disableEncounters = false;

            walkTime = 150;

            invisible = false;
            guard = false;

        }

        // DirectX functions and data members
        static public FontSurface Font { get { return myFont; } }					// returns the handle to the font resource
        static public Surface Tiles { get { return myTiles; } }				// returns the handle to the tiles resource
        static public Surface Character { get { return myCharacter; } }		// returns the handle to the character resource
        static public Surface Monsters { get { return pOverlandMonsters; } }				// returns the handle to the monsters resource
        static public Surface wallTexture;			// stores the pointer to the wall texture
        static public Surface floorTexture;			// stores the pointer to the floor texture
        static public Surface ceilingTexture;			// stores the pointer to the ceiling texture
        static public Surface floorHoleTexture;		// stores the pointer to the floorhole texture


        // character functions
        

        static Timing.StopWatch animWatch = new Timing.StopWatch();
        const int frameTime = 150;

        static public int AnimFrame
        {
            get 
            {
                int oldAnim = animFrame;

                if (animWatch.IsPaused == false)
                    animFrame = (((int)animWatch.TotalMilliseconds) / frameTime) % 3;

                if (oldAnim != animFrame)
                {
                    charAnimCount++;

                    if (charAnimCount > 6)
                    {
                        animFrame = 0;
                        charAnimCount = 0;
                        Animating = false;
                    }
                }

                return animFrame;
            }
            set
            {
                animFrame = value;

                while (animFrame < 0)
                    animFrame += 3;

                if (animFrame >= 3)
                    animFrame = value % 3;
            
                animWatch.Reset();
            }
        }
        /// <summary>
        /// sets or returns whether or not the character is animating
        /// </summary>
        /// <returns></returns>
        static public bool Animating
        {
            get
            {
                if (animating == false)
                {
                    animFrame = 0;
                }

                return animWatch.IsPaused == false;
            }
            set
            {
                if (value == false)
                {
                    animFrame = 0;
                    charAnimCount = 0;

                    if (animWatch.IsPaused == false)
                        animWatch.Pause();
                }
                else
                    animWatch.Resume();

            }
        }




        // menu functions
  


        
        static public SubMenu subMenu;					// the submenu
        static public bool LeftMenuActive;				// is the left menu active?
        static public Color HPColor;					// color of left status display

        // action window functions
        /// <summary>
        /// This function adds a line to the bottom of the action window and	
        /// scrolls the remaining lines up one each.
        /// </summary>
        /// <param name="?"></param>
        static public void AddBottom(ColorStringBuilder builder)
        {
            AddBottom(builder.Text, builder.Colors);
        }

        static public void AddBottom()
        {
            AddBottom();
        }
        static public void AddBottom(string line)
        {
            AddBottom(line, null);
        }
        static public void AddBottom(string line, Color color)
        {
            Color[] colors = new Color[40];

            for (int i = 0; i < 40; i++)
            {
                colors[i] = color;
            }

            AddBottom(line, colors);
        }

        static public Color bottomTextColor = XleColor.White;

        // adds a line to the bottom of the action window
        static public void AddBottom(string line, Color[] colors)
        {

            int i;

            bottom[4] = "";

            for (i = 3; i >= 0; i--)
            {
                bottom[i + 1] = bottom[i];
                bottomColor[i + 1] = bottomColor[i];
            }


            bottomColor[0] = new Color[39];

            bottom[0] = line;

            if (colors != null)
            {
                for (i = 0; i < 39; i++)
                {
                    Color clr;

                    if (i >= colors.Length)
                        clr = colors[colors.Length - 1];
                    else
                        clr = colors[i];

                    bottomColor[0][i] = clr;
                }
            }
            else
            {
                for (i = 0; i < 39; i++)
                {
                    bottomColor[0][i] = bottomTextColor;
                }
            }

        }
        static public void UpdateBottom(string line)
        {
            UpdateBottom(line, 0, null);
        }
        static public void UpdateBottom(string line, int loc)
        {
            UpdateBottom(line, loc, null);
        }
        static public void UpdateBottom(ColorStringBuilder builder, int loc)
        {
            UpdateBottom(builder.Text, loc, builder.Colors);
        }

        /// <summary>
        /// This function updates a line in the action window.
        /// </summary>
        static public void UpdateBottom(string line, int loc, Color[] colors)
        {
            bottom[loc] = line;

            if (colors != null)
            {
                for (int i = 0; i < 39; i++)
                {
                    Color clr;

                    if (i < colors.Length)
                        clr = colors[i];
                    else
                        clr = colors[colors.Length - 1];

                    bottomColor[loc][i] = clr;
                }

            }
        }


        static public void UpdateBottom(string line, Color color)
        {
            Color[] colors = new Color[40];

            for (int i = 0; i < 40; i++)
            {
                colors[i] = color;
            }

            UpdateBottom(line, 0, colors);

        }

        static public string Bottom(int line)
        {
            string tempspace;

            if (line >= 0 && line <= 4)
            {
                tempspace = bottom[line];

                return tempspace;
            }
            else
            {
                return null;
            }

        }


        static public Color[] BottomColor(int line)
        {

            Color[] tempspace = new Color[40];

            if (line >= 0 && line <= 4)
            {
                bottomColor[line].CopyTo(tempspace, 0);

                return tempspace;
            }
            else
            {
                return null;
            }

        }


        static public void ClearBottom()
        {
            int i;

            for (i = 0; i < 5; i++)
            {
                AddBottom("");
            }
        }

        static public void WriteSlow(string line, int loc, Color color)
        {


            int i = 0;
            Color[] colors = new Color[40];
            string temp;

            while (i < line.Length && i < 40)
            {
                colors[i++] = color;
                temp = line.Substring(0, i);

                UpdateBottom(temp, loc, colors);

                XleCore.wait(50);
            }

        }


        // others:
        static public bool LoadFont()
        {

            myFont = FontSurface.BitmapMonospace("font.png", new Size(16, 16));
            myFont.StringTransformer = StringTransformer.ToUpper;

            myCharacter = new Surface("character.png");
            pOverlandMonsters = new Surface("OverworldMonsters.png");

            
            return true;

        }
        static public bool LoadTiles(string tileset)
        {
            myTiles = new Surface(tileset);

            return true;
        }
        [Obsolete("Not needed, but loads strings and shit too.")]
        static public void SetHInstance()
        {

            /****************************************************************************
             *  void Global::SetHInstance(HINSTANCE instance)							*
             *																			*
             *  Here the hInstance value is stored for hInstance to return it when		*
             *	called.																	*
             ****************************************************************************/
            //void Global::SetHInstance(HINSTANCE instance, HWND window)
            //{
            //    if (locked == false)
            //    {
            //        char	tempSpace[40];
            //        int		i;

            //        myApp = instance;
            //        hWnd = window;

            //        for (i = 0; i < 8; i++)
            //        {
            //            LoadString (hInstance(), i + 1, tempSpace, 39);
            //            weaponName[i] = tempSpace;
            //        }

            //        for (i = 0; i < 5; i++)
            //        {
            //            LoadString (hInstance(), i + 9, tempSpace, 39);
            //            armorName[i] = tempSpace;

            //        }

            //        for (i = 0; i < 5; i++)
            //        {
            //            LoadString (hInstance(), i + 14, tempSpace, 39);
            //            qualityName[i] = tempSpace;

            //        }
            //    }
            //}
        }
        
        [Obsolete("This releases everything.")]
        static public void ReleaseFont()
        {

            /*
            LPDIRECTDRAWSURFACE7		myFont;				// stores the handle to the font
            LPDIRECTDRAWSURFACE7		myTiles;			// stores the handle to the tiles
            LPDIRECTDRAWSURFACE7		myCharacter;		// stores the handle to the character sprites
            LPDIRECTDRAWSURFACE7		pOverlandMonsters;	// stores the handle to the overland monster sprites

            LPDIRECTDRAWSURFACE7	wallTexture;			// stores the pointer to the wall texture
            LPDIRECTDRAWSURFACE7	floorTexture;			// stores the pointer to the floor texture
            LPDIRECTDRAWSURFACE7	ceilingTexture;			// stores the pointer to the ceiling texture
            LPDIRECTDRAWSURFACE7	floorHoleTexture;		// stores the pointer to the floorhole texture

            LPDIRECTINPUTDEVICE2	pJoystick;				// Joystick

            */

            if (myFont != null)
            {
                myFont.Dispose();
                myFont = null;
            }
            if (myTiles != null)
            {
                myTiles.Dispose();
                myTiles = null;
            }
            if (myCharacter != null)
            {
                myCharacter.Dispose();
                myCharacter = null;
            }
            if (pOverlandMonsters != null)
            {
                pOverlandMonsters.Dispose();
                pOverlandMonsters = null;
            }
            if (wallTexture != null)
            {
                wallTexture.Dispose();
                wallTexture = null;
            }
            if (floorTexture != null)
            {
                floorTexture.Dispose();
                floorTexture = null;
            }
            if (ceilingTexture != null)
            {
                ceilingTexture.Dispose();
                ceilingTexture = null;
            }
            if (floorHoleTexture != null)
            {
                floorHoleTexture.Dispose();
                floorHoleTexture = null;
            }

        }


        
        static public string WeaponName(int a) { return weaponName[a - 1]; }
        static public int WeaponCost(int w, int q)
        {
            return (int)(9.639302862 + 12.709725901 * w + 7.448174718 * Math.Pow(w, 2) +
                5.552075007 * q + 0.731199405 * Math.Pow(q, 2) + 4.290595417 * w * q);

        }
        static public string ArmorName(int a)
        {
            return armorName[a - 1];
        }
        static public int ArmorCost(int a, int q)
        {
            return 120 * a + 12 * q + 8 * q * a;
        }
        static public string QualityName(int a)
        {
            return qualityName[a];
        }


        // map:
        //static public LotaMap map;							// the map class

        // other commonly used variables that don't need accessors
        static public char currentCommand;			// the first letter of the current command being executed
        static public int waitCommand;			// holds the time in msec to wait to give Enter Command
        static public int walkTime;				// time to wait between steps

        static public int screenLeft;				// these two values store the original position
        static public int screenTop;				//		of the window so we can restore it upon exiting 
        //		full screen mode
        static public int stdDisplay;				// heartbeat has control of the display when = 0
        static private bool done;

        public static bool Done
        {
            get
            {
                if (Display.CurrentWindow.IsClosed)
                    return true;
                else
                    return g.done;
            }
            set { g.done = value; }
        }

        static public bool waterReset;				// reset the water dots
        static public int vertLine;				// Vertical line dividing menu and map
        static public bool newGraphics;			// are we displaying the new graphics?
        static public int raftAnim;				// raft animation frame
        static public Direction raftFacing;		// direction the raft is facing (lotaEast or lotaWest)
        static public int charAnimCount;			// animation count for the player
        static public bool ZBufferEnable;			// enables or disables the ZBuffer

        static public bool d3dViewport;			// stores whether or not the d3d viewport has been set
        //ErikCollection<D3DFloor*> d3dFloor;		// first wall of the d3d area
        //static public D3DFloor* d3dFloor;
        static public bool allowEnter;				// used when a player exits a town or dungeon
        static public bool invisible;				// is the player invisible?
        static public bool guard;					// is the player in guard colors?

        static public bool disableEncounters;		// used to disable overworld encounters


        internal static bool HasCommand(KeyCode cmd)
        {
            return true;
        }
    }
}